File: Assignment14/Q1.c
///////////////////////////////////////////////////////////////////
//
//      Including Header Files
//
///////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>

///////////////////////////////////////////////////////////////////
//
//      Function name:  CountEven
//      Input:          Integer Array, Integer
//      Output:         Integer
//      Description:    Accept array and return frequency
//                      of even numbers
//      Date:           03/06/2026
//      Author:         Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

int CountEven(
                int Arr[],     // Input array
                int iLength    // Number of elements
             )
{
    int iCnt = 0;              // Loop counter
    int iCount = 0;            // Stores frequency of even numbers

    // Traverse array
    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        // Check whether element is even
        if((Arr[iCnt] % 2) == 0)
        {
            iCount++;
        }
    }

    return iCount;
}

///////////////////////////////////////////////////////////////////
//
//      Entry point function
//
///////////////////////////////////////////////////////////////////

int main()
{
    int iSize = 0;             // Stores number of elements
    int iRet = 0;              // Stores return value from function
    int iCnt = 0;              // Loop counter

    int *p = NULL;             // Pointer for dynamic memory allocation

    printf("Enter number of elements\n");
    scanf("%d",&iSize);

    // Allocate memory dynamically
    p = (int *)malloc(iSize * sizeof(int));

    // Check whether memory allocation is successful
    if(p == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    printf("Enter %d elements\n",iSize);

    // Accept elements from user
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element : %d\n",iCnt + 1);
        scanf("%d",&p[iCnt]);
    }

    // Function call
    iRet = CountEven(p, iSize);

    // Display result
    printf("Result is %d\n",iRet);

    // Deallocate memory
    free(p);

    return 0;
}

///////////////////////////////////////////////////////////////////
//
//      Input :
//      N : 6
//
//      Elements :
//      85 66 3 80 93 88
//
//      Output :
//      3
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Where n is the number of elements in array
//
///////////////////////////////////////////////////////////////////

File: Assignment20/Q1.java

///////////////////////////////////////////////////////////////////
//
//      File Name : Q1.java
//      Project   : Logic Building Assignment 20
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : EvenSumLogic
//      Method Name: sumEvenNumbers
//      Input      : Integer
//      Output     : Void
//      Description: Find the sum of all even numbers
//                   up to N
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class EvenSumLogic
{
    void sumEvenNumbers(int n)
    {
        int i = 0;      // Loop counter
        int iSum = 0;   // Store summation

        // Calculate sum of even numbers
        for(i = 1; i <= n; i++)
        {
            if(i % 2 == 0)
            {
                iSum = iSum + i;
            }
        }

        // Display result
        System.out.println("Summation of even numbers is : " + iSum);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q1
{
    public static void main(String args[])
    {
        // Create object of EvenSumLogic class
        EvenSumLogic obj = new EvenSumLogic();

        // Function call
        obj.sumEvenNumbers(10);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 10
//
//      Output :
//      Summation of even numbers is : 30
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////
